server_tests/test/clientTests.js

```diff
@@ -791,514 +736,468 @@ describe("RUST API TEST SUITE", function() {
                 Comparison.oscCloseSessionOutput(res);
-                done();
-            }));
+            }).catch(shouldNotGetHere);
         });
 
-        it("Expect success. camera.getOptions gets correct options when gettable options are set to supported values", function(done) {
-            testClient.getOptions(sessionId, specifiedOptions)
-            .then( Comparison.catchExceptions(done, function(res) {
+        it("Expect success. camera.getOptions gets correct options when gettable options are set to supported values", function() {
+            return testClient.getOptions(sessionId, specifiedOptions)
+            .then( function onSuccess (res) {
                 Comparison.oscGetOptionsOutput(specifiedOptions, res);
-                done();
-            }));
+            }, shouldNotGetHere);
         });
 
-        it("Expect missingParameter Error. camera.getOptions cannot get options when options is not provided", function(done) {
-            testClient.getOptions(sessionId)
-            .then( Comparison.catchExceptions(done, function(res) {
-                Comparison.missingParameterError(res);
-                done();
-            }));
+        it("Expect missingParameter Error. camera.getOptions cannot get options when options is not provided", function() {
+            return testClient.getOptions(sessionId)
+            .then( shouldNotGetHere, function onError (err) {
+                Comparison.missingParameterError(err);
+            });
         });
 
-        it("Expect missingParameter Error. camera.getOptions cannot get options when sessionId is not provided", function(done) {
-            testClient.getOptions(undefined, specifiedOptions)
-            .then( Comparison.catchExceptions(done, function(res) {
-                Comparison.missingParameterError(res);
-                done();
-            }));
+        it("Expect missingParameter Error. camera.getOptions cannot get options when sessionId is not provided", function() {
+            return testClient.getOptions(undefined, specifiedOptions)
+            .then( shouldNotGetHere, function onError (err) {
+                Comparison.missingParameterError(err);
+            });
         });
 
         // RE-ADD ONCE EXTRA FIELD CHECKING HAS BEEN IMPLEMENTED
-        it.skip("Expect invalidParameterValue Error. camera.getOptions cannot get options when options is set to unsupported value", function(done) {
-            testClient.getOptions(sessionId, ['wrongtype'])
-            .then( Comparison.catchExceptions(done, function(res) {
-                Comparison.invalidParameterValueError(res);
-                done();
-            }));
+        it.skip("Expect invalidParameterValue Error. camera.getOptions cannot get options when options is set to unsupported value", function() {
+            return testClient.getOptions(sessionId, ['wrongtype'])
+            .then( function onSuccess (res) {
+              asser.fail();
```

Comment:
Misspelled `assert`, and also inconsistent with previous usage of `shouldNotGetHere` in place of explicit assert.